client/app/src/main/java/com/microsoft/research/karya/ui/registration/ProfilePictureFragment.kt

```diff
@@ -36,16 +68,29 @@ class ProfilePictureFragment : Fragment(R.layout.fragment_profile_picture) {
         /** Initialise assistant audio **/
         registrationActivity.current_assistant_audio = R.string.audio_profile_picture_prompt
 
-        binding.mainProfilePictureIv.setOnClickListener { getProfilePicture() }
+        with(binding) {
 
-        binding.profilePictureNextIv.setOnClickListener {
-            binding.mainProfilePictureIv.isClickable = false
-            binding.rotateRightIb.isClickable = false
-            binding.profilePictureNextIv.visibility = View.INVISIBLE
-            submitProfilePicture()
-        }
+            binding.mainProfilePictureIv.setOnClickListener { getProfilePicture() }
+
+            binding.profilePictureNextIv.setOnClickListener {
+                binding.mainProfilePictureIv.isClickable = false
+                binding.rotateRightIb.isClickable = false
+                binding.profilePictureNextIv.visibility = View.INVISIBLE
+                val imageFolder = requireActivity().getDir("profile_picture", AppCompatActivity.MODE_PRIVATE).path
+
+                if (::profilePic.isInitialized) {
+                    viewModel.submitProfilePicture(profilePic, imageFolder)
+                } else {
+                    navigateToSelectGenderFragment()
+                }
+            }
 
-        binding.rotateRightIb.setOnClickListener { rotateRight() }
+            binding.rotateRightIb.setOnClickListener {
+                if (::profilePic.isInitialized) {
+                    profilePic = viewModel.rotateRight(profilePic)
+                }
+            }
+        }
```

Comment:
Why use the `with` scope function if the `this` context object is not being used?